src/types/notification.ts

```diff
@@ -1,12 +1,75 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {NotificationConstructorOptions} from 'electron/common';
+import {Notification, NotificationMetadata} from 'node-notifier';
+import WindowsBalloon from 'node-notifier/notifiers/balloon';
+import Growl from 'node-notifier/notifiers/growl';
+import NotifySend from 'node-notifier/notifiers/notifysend';
+import WindowsToaster from 'node-notifier/notifiers/toaster';
 
-export type MentionData = {
+export type NotificationOptions = WindowsToaster.Notification & WindowsBalloon.Notification & NotifySend.Notification & Growl.Notification & Notification;
+
+type MentionPayload = {
+    channel: {
```

Comment:
(nitpick) perhaps channel can be its own type, since it's used a couple of times (here and in `DisplayMentionArguments`)